sys/vane/jael.hoon

```diff
@@ -1595,11 +2051,24 @@
     ::
     ::TODO  if the block number is less than latest, that means we got
     ::      events out of order somehow and should probably reset.
-    ?>  (gth block-number.place latest-block)
+    ::      This could also mean there was a chain reorg if the logs
+    ::      have the 'removed' tag set.  In this case, we should delete
+    ::      the old logs.  Finally, since we rewind 40 blocks on new
+    ::      filter, this could be up to 40 blocks old just because of
+    ::      that.
+    ::  ~?  (lte block-number.place latest-block)
+    ::    [%old-block block-number.place latest-block]
     ::
     ?:  (~(has in heard) block-number.place log-index.place)
-      ~&  %ignoring-duplicate-event
-      +>
+      ?.  removed.u.mined.log
+        ::  ~&  [%ignoring-duplicate-event tx=transaction-hash.u.mined.log]
+        +>
+      ~&  :*  'removed event!  Perhaps chain has reorganized?'
+              tx-hash=transaction-hash.u.mined.log
+              block-number=block-number.u.mined.log
+              block-hash=block-hash.u.mined.log
+          ==
+      +>  ::TODO  undo the effects of this event
```

Comment:
As above, I believe we made the changes necessary to support appropriate behavior here. Can we start actually using that in the logic?